app/src/main/java/com/keylesspalace/tusky/fragment/ViewImageFragment.kt

```diff
@@ -131,49 +132,74 @@ class ViewImageFragment : ViewMediaFragment() {
         super.onDestroyView()
     }
 
-    private fun loadImageFromNetwork(url: String, photoView: ImageView) =
-            //Request image from the any cache
-            Glide.with(this)
-                    .load(url)
-                    .dontAnimate()
-                    .onlyRetrieveFromCache(true)
-                    .error(
-                            //Request image from the network on fail load image from cache
-                            Glide.with(this)
-                                    .load(url)
-                                    .centerInside()
-                                    .addListener(ImageRequestListener(false))
-                    )
-                    .centerInside()
-                    .addListener(ImageRequestListener(true))
-                    .into(photoView)
-
+    private fun loadImageFromNetwork(url: String, previewUrl: String?, photoView: ImageView) {
+        val glide = Glide.with(this)
+        // Request image from the any cache
+        glide
+                .load(url)
+                .dontAnimate()
+                .onlyRetrieveFromCache(true)
+                .let {
+                    if (previewUrl != null)
+                        it.thumbnail(glide
+                                .load(previewUrl)
+                                .dontAnimate()
+                                .onlyRetrieveFromCache(true)
+                                .centerInside()
+                                .addListener(ImageRequestListener(true, isThumnailRequest = true)))
+                    else it
+                }
+                //Request image from the network on fail load image from cache
+                .error(glide.load(url)
+                        .centerInside()
+                        .addListener(ImageRequestListener(false, isThumnailRequest = false))
+                )
+                .centerInside()
+                .addListener(ImageRequestListener(true, isThumnailRequest = false))
+                .into(photoView)
+    }
 
     /**
      * @param isCacheRequest - is this listener for request image from cache or from the network
      */
-    private inner class ImageRequestListener(private val isCacheRequest: Boolean) : RequestListener<Drawable> {
-        override fun onLoadFailed(e: GlideException?, model: Any?, target: Target<Drawable>?, isFirstResource: Boolean): Boolean {
-            if (isCacheRequest) //Complete the transition on failed image from cache
-                completeTransition()
-            else
-                progressBar?.hide() //Hide progress bar only on fail request from internet
+    private inner class ImageRequestListener(
+            private val isCacheRequest: Boolean,
+            private val isThumnailRequest: Boolean) : RequestListener<Drawable> {
```

Comment:
Can't you use `isFirstResource` instead of this custom boolean? Shouldn't it be the same?

Reply:
If there is a thumbnail. There might be none.